test/DropdownButtonSpec.js

```diff
@@ -486,19 +517,27 @@ describe('DropdownButton', function() {
       buttonNode.getAttribute('aria-expanded').should.equal('true');
     });
 
-    it('when open and the key "esc" is pressed the menu is closed and focus is returned to the button', function() {
+    // I am not sure why this test has to be so convoluted.
+    // the naive approach would fail _only_ if the entire test suite was run together, but I couldn't isolate it any more
```

Comment:
hehe, sorry, my quick attempt to explain why this test is more complicated looking then it seems like it should be. 

Essentially the test as it was, would sporadically fail, when the entire test suite was run. It would pass with flying colors when just that test, or even all the dropdown tests, were run. I tried to isolate which tests were causing the conflict but didn't have any luck. I reworked the test, so it doesn't sometimes fail (it still makes the same assertions).

I feel fairly confident that the failure was due to a test specific conflict and not an actual bug.